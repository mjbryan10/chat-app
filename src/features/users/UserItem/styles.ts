import styled, { ThemeProps, Theme } from 'styled-components';

export const Container = styled.div<ThemeProps<Theme>>`
  display: flex;
  align-items: center;
  justify-content: center;
  border: ${({theme}) => theme.border}
`;