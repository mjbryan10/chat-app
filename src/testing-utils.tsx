import React from 'react';
import { render } from '@testing-library/react'
import { ThemeProvider } from 'styled-components'
import { dark } from './shared/theme';

const AllTheProviders = ({ children }: any) => {
  return (
    <ThemeProvider theme={dark}>
        {children}
    </ThemeProvider>
  )
  
}

const customRender = (ui?: any, options?: any) =>
  render(ui, { wrapper: AllTheProviders, ...options })

// re-export everything
export * from '@testing-library/react'

// override render method
export { customRender as render }