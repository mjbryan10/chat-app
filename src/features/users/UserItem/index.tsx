import React, { FC, SyntheticEvent } from 'react';
import * as S from './styles';
import { User } from 'shared/Api/types';

interface Props {
  user: User;
  handleClick?: (id: number) => void;
}

const UserItem: FC<Props> = ({user, handleClick}) => {

  const onClick = (id: number) => (event: SyntheticEvent<HTMLElement>) => {
    event.preventDefault();
    if (handleClick) handleClick(id);
  }
  return (
    <S.Container onClick={onClick(user.id)} data-testid="user-item">
      {user.name}
    </S.Container>
  )
}

export default UserItem
