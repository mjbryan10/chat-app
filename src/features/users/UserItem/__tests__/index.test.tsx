import React from 'react';
import { RenderResult, render } from 'testing-utils';
import UserItem from '..';
import { cleanup, fireEvent } from '@testing-library/react';

describe('UserItem Feature Component', () => {
  const __user = {id: 1, name: 'Wessel'}
  const __handleClick = jest.fn();
  let utils: RenderResult;
  beforeEach(() => 
    utils = render(<UserItem user={__user} handleClick={__handleClick}/>)
  )
  afterEach(() => {
    cleanup();
  })
  test('should display the user name from props', () => {
    const {getByText} = utils;

    expect(getByText(/Wessel/i)).toBeVisible();
  })

  test('should call click handler when item is clicked', () => {
    const { getByTestId } = utils;

    fireEvent.click(getByTestId('user-item'));

    expect(__handleClick).toBeCalledTimes(1);
  })
  
  
})
