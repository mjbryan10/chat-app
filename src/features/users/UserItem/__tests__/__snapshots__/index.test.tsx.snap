// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserItem Feature Component should render correctly 1`] = `
<DocumentFragment>
  <div
    class="sc-AxjAm fNKPyA"
    data-testid="user-item"
  >
    Wessel
  </div>
</DocumentFragment>
`;
